packages/api/src/task/scheduler.ts

```diff
@@ -378,23 +379,27 @@ export class TaskScheduler {
   async createTask(
     type: Task["type"],
     params: Task["params"],
+    projectId: string,
     inputAsset?: Asset,
     outputAsset?: Asset,
     userId?: string,
     requesterId?: string,
   ) {
+    const uId = inputAsset?.userId || outputAsset?.userId || userId;
+    const user = await db.user.get(uId, { useCache: true });
```

Comment:
Should not be re-fetching the user everywhere. Can you just pass the object around?